### PR TITLE
Tolerate invalid symbol indices.

### PR DESCRIPTION
**Pull request by matthias-blume**
These can appear in relocations related to debug info.

Also, print the name of the offending symbol if it refers to an invalid segment.

### CODE REVIEW

**Comment by matthias-blume** [comment]
I'm honestly a bit stumped right now about those build failures.  Builds fine locally...

File: src/binary-reader-ir.cc [hunk 1]
```diff
@@ -1619,6 +1619,9 @@ Result BinaryReaderIR::OnReloc(RelocType type,
   if (options_.features.sandbox_enabled()) {
     return Result::Ok;
   }
+  if (index == kInvalidIndex) {
```
Comment (dimitry-) [inline src/binary-reader-ir.cc]: maybe add a comment why we are doing this.. Ideally also check that this is indeed a debug relocation.

**Review by dimitry- — APPROVED** [review]

File: src/binary-reader-ir.cc (hunk 1 above)
Comment (matthias-blume) [reply src/binary-reader-ir.cc]: I added a comment, including a link to the code that Sam showed us where these indices are generated.
I changed things so we tolerate -1 and also -2.
Checking the section name might be valuable, but it's a bit messy and we need to change the OnReloc interface to plumb the name through.  So I decided to not do this for now.  (This is only in the no-sandbox branch anyway.)

**Comment by matthias-blume** [comment]
> I'm honestly a bit stumped right now about those build failures. Builds fine locally...

It still fails with mysterious symptoms that I don't see locally.
Any idea what might be going on there.  It seems completely unrelated to the change.

**Comment by sbc100** [comment]
> > I'm honestly a bit stumped right now about those build failures. Builds fine locally...
> 
> It still fails with mysterious symptoms that I don't see locally. Any idea what might be going on there. It seems completely unrelated to the change.

Are the tests passing on main?  Perhaps an merge of recent upstream changes might help?

**Comment by matthias-blume** [comment]
> Are the tests passing on main? Perhaps an merge of recent upstream changes might help?

No, they don't (I think).  I just tried a test PR that only changed a single line, adding a comment.  It had the same problem.  So I think there's something wrong with main.  I wonder how it got into that state...

**Comment by sbc100** [comment]
> > Are the tests passing on main? Perhaps an merge of recent upstream changes might help?
> 
> No, they don't (I think). I just tried a test PR that only changed a single line, adding a comment. It had the same problem. So I think there's something wrong with main. I wonder how it got into that state...

It could be that something changes with the github CI imagines.   I would try merging from upstream.

**Pull request merged**